packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -0,0 +1,491 @@
+};
+
+const isValidTransaction = (tx: ParsedTransactionWithMeta): tx is SolanaValidParsedTxWithMeta =>
+    !!(tx && tx.meta && tx.transaction && tx.blockTime);
+
+const pushTransaction = async (request: Request<MessageTypes.PushTransaction>) => {
+    const rawTx = request.payload.startsWith('0x') ? request.payload.slice(2) : request.payload;
+    const api = await request.connect();
+    const payload = await api.sendRawTransaction(Buffer.from(rawTx, 'hex'));
+    return {
+        type: RESPONSES.PUSH_TRANSACTION,
+        payload,
+    } as const;
+};
+
+const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) => {
+    const { payload } = request;
+    const { details = 'basic' } = payload;
+    const api = await request.connect();
+
+    const publicKey = new PublicKey(payload.descriptor);
+
+    const accountInfo = await api.getAccountInfo(publicKey);
+
+    if (!accountInfo) {
+        // return empty account
+        const account: AccountInfo = {
+            descriptor: payload.descriptor,
+            balance: '0', // default balance
+            availableBalance: '0', // default balance
+            empty: true,
+            history: {
+                total: -1,
+                unconfirmed: 0,
+                transactions: undefined,
+            },
+        };
+        return Promise.resolve({
+            type: RESPONSES.GET_ACCOUNT_INFO,
+            payload: account,
+        } as const);
+    }
+
+    const getTransactionPage = async (txIds: string[]) => {
+        const transactionsPage = await fetchTransactionPage(api, txIds);
+        const uniqueTransactionSlots = Array.from(new Set(transactionsPage.map(tx => tx.slot)));
+
+        // we do not get blockheight from the transaction history, we only get slot for each transaction.
+        // for this reason we have to fetch block for each slot and an create a dictionary
+        const slotToBlockHeightMapping = (
+            await Promise.all(
+                uniqueTransactionSlots.map(async slot => {
+                    const { blockHeight } = await api.getParsedBlock(slot, {
+                        maxSupportedTransactionVersion: 0,
+                    });
+                    return blockHeight;
+                }),
+            )
+        ).reduce(
+            (acc, curr, i) => ({ ...acc, [uniqueTransactionSlots[i]]: curr }),
+            {} as Record<number, number | null>,
+        );
+
+        return transactionsPage
+            .filter(isValidTransaction)
+            .map(tx =>
+                solanaUtils.transformTransaction(tx, payload.descriptor, slotToBlockHeightMapping),
+            )
+            .filter((tx): tx is Transaction => !!tx);
+    };
+
+    const tokenAccounts = await api.getParsedTokenAccountsByOwner(publicKey, {
+        programId: new PublicKey(TOKEN_PROGRAM_PUBLIC_KEY),
+    });
+
+    const allAccounts = [payload.descriptor, ...tokenAccounts.value.map(a => a.pubkey.toString())];
```

Comment:
I guess Stake Accounts are not part of this PR or out of scope, right?

Reply:
Yes, staking is out of scope for now.